File: eternalxi_front/lib/core/utils/user_public_tag.dart
/// Tag publico unico derivado del id de usuario (p. ej. #623423).
abstract final class UserPublicTag {
  UserPublicTag._();

  static const _multiplier = 7919;
  static const _offset = 104729;
  static const _mod = 900000;
  static const _base = 100000;

  static int codeForUserId(int userId) {
    if (userId <= 0) return _base;
    return _base + ((userId * _multiplier + _offset) % _mod);
  }

  static String format(int userId) => '#${codeForUserId(userId)}';

  static int? parseCode(String raw) {
    var value = raw.trim();
    if (value.startsWith('#')) {
      value = value.substring(1);
    }
    if (!RegExp(r'^\d{5,7}$').hasMatch(value)) {
      return null;
    }
    return int.tryParse(value);
  }
}

File: eternalxi_api_back/src/main/java/com/eternalxi/eternalxi_api/util/UserPublicTag.java
package com.eternalxi.eternalxi_api.util;

public final class UserPublicTag {

    private static final int MULTIPLIER = 7919;
    private static final int OFFSET = 104729;
    private static final int MOD = 900000;
    private static final int BASE = 100000;

    private UserPublicTag() {}

    public static int codeForUserId(long userId) {
        if (userId <= 0) {
            return BASE;
        }
        return BASE + (int) ((userId * MULTIPLIER + OFFSET) % MOD);
    }

    public static Long userIdFromTagCode(int tagCode) {
        if (tagCode < BASE || tagCode >= BASE + MOD) {
            return null;
        }
        int target = tagCode - BASE;
        for (long id = 1; id <= 5_000_000L; id++) {
            if (((id * MULTIPLIER + OFFSET) % MOD) == target) {
                return id;
            }
        }
        return null;
    }
}
